### ISSUE TITLE
[BUG] Using the "map <map number>" command in terminal on servers with a map list confuses "nextmap".

### ISSUE THREAD

**Issue by Acts19quiz** [odamex/odamex#828]
**Describe the bug**
If a server is configured with a maplist (e.g., "addmap MAP##"), using the "map" command in the server terminal will warp from the current map to the specified map. _**However**_, the next invocation of "nextmap" (either from terminal or from a callvote) will not go to the next map according to the current map. Rather, it will instead go to the next map specified after the map the "map" terminal command was used on.

So say the maplist is:
```
addmap MAP01 DOOM2.WAD
addmap MAP02
addmap MAP03
addmap MAP04
addmap MAP05
```
If the game advances to MAP02, the admin sees an issue (or has a preference), and puts in "map MAP04" into the terminal, then during MAP04 the players call a successful vote for "nextmap", the next map will not be MAP05, but MAP03, because some internal check still thinks MAP02 is what to check against in determining what the nextmap truly is.

**Build that the bug occurred in**
10.3.0 GB28EE-8578

**To Reproduce**
1. Start a server configured with a map list.
2. In the server terminal, use the "map <map number>" command to go directly to a map in the list.
3. Use "nextmap", either in the terminal or as "callvote nextmap".

**Expected behavior**
For "nextmap" to go to whatever is the next map after the current actual map in-progress, and for the "map <map number>" command to not bork things.

**Screenshots, NetDemos, & Crash Dumps**
N/A

**Additional context**
This doesn't seem to affect servers that don't have a map list.

**Comment by rakohus** [comment]
I'm not sure this is a bug at all. This scenario presumes that the server admin only wants to stay within the existing maplist with the `map` command, but they are not limited to what is in the maplist. For instance, here the server admin could pass `map map10`, which is not in the list. Then what happens? The admin also has a `nextmap` ccmd available to them if they want to iterate past this map in the list. 

I see the angle you're getting at where you might want to jump several maps ahead instead of one, but there are corner cases to consider, especially when the maplist may contain entirely other wads in the order as well.

**Comment by Acts19quiz** [comment]
What does nextmap do in cases where no map list is defined?

**Comment by loopfz** [comment]
If no maplist is defined, `nextmap` progresses naturally through the wad.

In the scenario you described where the admin wants to move around within the confines of the maplist, I think the correct command to use would be `gotomap`. Then `callvote nextmap` would function as described.

I think the functionality is fine, but this would warrant a wiki section perhaps?

**Comment by rakohus** [comment]
> If no maplist is defined, `nextmap` progresses naturally through the wad.
> 
> In the scenario you described where the admin wants to move around within the confines of the maplist, I think the correct command to use would be `gotomap`. Then `callvote nextmap` would function as described.
> 
> I think the functionality is fine, but this would warrant a wiki section perhaps?

Yes, this is likely an area for documentation over changing any of the current behaviors
